integrations/dell_idrac/dell_idrac_test.py

```diff
@@ -0,0 +1,203 @@
+        with self.assertRaises(SaasException) as ctx:
+            plugin.rollback_password()
+        self.assertIn("Rollback failed", str(ctx.exception))
+
+
+
+class DelliDRACClientTest(unittest.TestCase):
+
+    def setUp(self):
+        user = MagicMock()
+        user.username.value = "testuser"
+        self.client = DelliDRACClient(
+            admin_username="admin",
+            admin_password="adminpass",
+            idrac_ip="1.2.3.4",
+            user=user
+        )
+
+    @patch("dell_idrac.requests.get")
+    def test_check_username_by_id_success(self, mock_get):
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {"UserName": "testuser"}
+        mock_get.return_value = mock_response
+
+        # Should not raise
+        self.client.check_username_by_id("42")
+
+    @patch("dell_idrac.requests.get")
+    def test_check_username_by_id_mismatch(self, mock_get):
+        mock_response = MagicMock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {"UserName": "otheruser"}
+        mock_get.return_value = mock_response
+
+        with self.assertRaises(SaasException) as ctx:
+            self.client.check_username_by_id("42")
+        self.assertIn("Username mismatch with user_id", str(ctx.exception))
+
+    @patch("dell_idrac.requests.get")
+    def test_check_username_by_id_not_found(self, mock_get):
+        mock_response = MagicMock()
+        mock_response.status_code = 404
+        mock_get.return_value = mock_response
+
+        with self.assertRaises(SaasException) as ctx:
+            self.client.check_username_by_id("42")
+        self.assertIn("User not present in Dell iDRAC", str(ctx.exception))
+
+    @patch("dell_idrac.requests.get")
+    def test_check_username_by_id_unexpected_error(self, mock_get):
+        mock_response = MagicMock()
+        mock_response.status_code = 500
+        mock_response.text = "Internal Server Error"
+        mock_get.return_value = mock_response
+
+        with self.assertRaises(SaasException) as ctx:
+            self.client.check_username_by_id("42")
+        self.assertIn("Unexpected error: 500", str(ctx.exception))
+
+    @patch("dell_idrac.requests.get", side_effect=Exception("Network error"))
+    def test_check_username_by_id_exception(self, mock_get):
+        with self.assertRaises(SaasException) as ctx:
+            self.client.check_username_by_id("42")
+        self.assertIn("Failed to verify user ID: Network error", str(ctx.exception))
+
+    @patch("dell_idrac.requests.patch")
+    def test_change_dell_idrac_user_password_success(self, mock_patch):
+        mock_response = MagicMock()
+        mock_response.status_code = 204
+        mock_patch.return_value = mock_response
+
+        # Should not raise
+        self.client.change_dell_idrac_user_password("42", "newpass")
```

Comment:
Even if it does not raise an exception, it should still check for the function calls.